### PR TITLE
refactor: Remove `aws-resource-tags` module and its usage from blueprints

### LINKED ISSUE #543: [REFACTOR] - Non-prescriptive tagging
#### Is your feature request related to a problem? Please describe

Not related to a problem or issue - but feedback from customers around tags.

#### Describe the solution you'd like

Currently, EKS Blueprints leverages the [aws-resource-tags](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/modules/aws-resource-tags) submodule for tags. We should be less prescriptive around the tagging scheme that should be used, and allow customers to define their own scheme. 

WIP branch is here https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/497

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?
1. Removes the sub-module `aws-resource-tags` and its usage through the blueprints/examples
2. Update examples to create three distinct sections
	- Provider & locals - mostly setup to execute the example
	- The blueprint definition (cluster and addons) 
	- Supporting resources - any resources that are not blueprint specific but required to make the example deployable. The resources contained in this section is assumed to be left up to users (create new like example, integrate with existing, etc.)
3. Various updates
	- For numeric values, use numbers instead of strings
	- Rename `aws_vpc` to just simply `vpc` 

### Motivation
1. Tags are typically very specific to a users environment and their current standards, practices, business requirements, etc. Instead of being overly prescriptive on tags within the blueprint definition, we can instead leave that up to users unless there is a strong reason that is directly related tied to blueprints. In addition, the tag module is also serving as a naming resource which again is enforcing a naming scheme upon users instead of letting users pick the naming scheme which can be problematic when conflicting with business/compliance requirements
	- Relates to #333 as well
2. Clarify how the information is presented to users to make it easier to see what portion is relevant to blueprints, and what is not (there to support deploying the example). This also makes it easier for users to copy+paste the blueprint section where only minimal changes are required to implement within their environment
3. Closes #543 

### More

- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: .github/workflows/e2e-parallel-full.yml [hunk 1]
```diff
@@ -52,11 +51,9 @@ jobs:
             tenant_name: 'nginx'
             deployment_order:
               [
-                'module.e2e_test.module.aws_vpc',
+                'module.e2e_test.module.vpc',
```
Comment (bryantbiggs) [inline .github/workflows/e2e-parallel-full.yml:54]: when I tested this example, it works with just a simple `terraform apply` but leaving this split out to be safe

File: examples/analytics/emr-on-eks/main.tf [hunk 2]
```diff
@@ -118,52 +80,77 @@ module "eks_blueprints" {
...
+#---------------------------------------------------------------
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 3.0"
+
+  name = local.name
+  cidr = "10.0.0.0/16"
+
+  azs             = ["${local.region}a", "${local.region}b", "${local.region}c"]
+  private_subnets = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24"]
```
Comment (kcoleman731) [inline examples/analytics/emr-on-eks/main.tf:138]: Why change from the previous approach?

File: examples/complete-kubernetes-addons/main.tf [hunk 3]
```diff
@@ -222,66 +176,47 @@ module "eks_blueprints_kubernetes_addons" {
...
+  enable_nat_gateway   = true
+  single_nat_gateway   = true
+  enable_dns_hostnames = true
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${local.name}" = "shared"
+    "kubernetes.io/role/elb"              = "1"
   }
 
-  enable_fargate_fluentbit = true
```
Comment (kcoleman731) [inline examples/complete-kubernetes-addons/main.tf:252]: do we have this in a different example?

File: examples/complete-kubernetes-addons/main.tf [hunk 4]
```diff
@@ -222,66 +176,47 @@ module "eks_blueprints_kubernetes_addons" {
 
   enable_amazon_eks_aws_ebs_csi_driver = true
 
-  # Prometheus and Amazon Managed Prometheus integration
-  enable_prometheus                    = true
-  enable_amazon_prometheus             = true
-  amazon_prometheus_workspace_endpoint = module.eks_blueprints.amazon_prometheus_workspace_endpoint
-
-  enable_aws_for_fluentbit = true
-  aws_for_fluentbit_helm_config = {
```
Comment (kcoleman731) [inline examples/complete-kubernetes-addons/main.tf:231]: do we have this in a different example?

File: examples/fully-private-eks-cluster/main.tf [hunk 5]
```diff
@@ -14,173 +14,120 @@ provider "kubernetes" {
...
+  cluster_endpoint_public_access  = true
+  cluster_endpoint_private_access = true
+
+  # Step 2. Change cluster endpoint to private only, comment out the above lines and uncomment the below lines.
+  # cluster_endpoint_public_access  = false
+  # cluster_endpoint_private_access = true
+
+  # EKS MANAGED NODE GROUPS
+  managed_node_groups = {
+    mg_4 = {
```
Comment (kcoleman731) [inline examples/fully-private-eks-cluster/main.tf:49]: mg_5?

File: .github/workflows/e2e-parallel-destroy.yml [hunk 6]
```diff
@@ -51,11 +50,11 @@ jobs:
             tenant_name: 'nginx'
             deployment_order:
               [
-                'module.e2e_test.module.aws_vpc',
+                'module.e2e_test.module.vpc',
                 'module.e2e_test.module.eks_blueprints',
                 'module.e2e_test.module.eks_blueprints_kubernetes_addons',
-                'module.e2e_test.module.aws_load_balancer_controller',
-                'module.e2e_test.module.ingress_nginx',
+                'module.e2e_test.module.eks_blueprints_kubernetes_addons.module.aws_load_balancer_controller[0]',
```
Comment (kcoleman731) [inline .github/workflows/e2e-parallel-destroy.yml:56]: Does this exclude LBC from being provisioned as part of `eks_blueprints_kubernetes_addons` target apply?

**Review by kcoleman731 — COMMENTED** [review]
Good stuff! Small comments.

File: examples/analytics/emr-on-eks/main.tf (hunk 2 above)
Comment (bryantbiggs) [reply examples/analytics/emr-on-eks/main.tf:138]: Just trying to reduce the amount of configs users need to copy+paste to replicate the blueprint in their current codebase. Functionally, it achieves the same thing but we're focusing more on the blueprints and less on VPC (which I assume will already exist for most users so lets keep that out of the "signal"). Does that answer your question?

### Before
image

### After
image

File: examples/complete-kubernetes-addons/main.tf (hunk 3 above)
Comment (bryantbiggs) [reply examples/complete-kubernetes-addons/main.tf:252]: Ah, thanks for catching this - this was not intended and will add it back in

File: .github/workflows/e2e-parallel-destroy.yml (hunk 6 above)
Comment (bryantbiggs) [reply .github/workflows/e2e-parallel-destroy.yml:56]: Thought I will defer to @Zvikan to make sure I am doing everything properly for the GitHub actions workflow

File: examples/fully-private-eks-cluster/main.tf (hunk 5 above)
Comment (bryantbiggs) [reply examples/fully-private-eks-cluster/main.tf]: Updated in https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/551/commits/3335bd466858a2527a90dad8256a0597a0a56262

File: examples/complete-kubernetes-addons/main.tf (hunk 4 above)
Comment (bryantbiggs) [reply examples/complete-kubernetes-addons/main.tf]: This wasn't intended - corrected in https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/551/commits/3335bd466858a2527a90dad8256a0597a0a56262

File: examples/complete-kubernetes-addons/main.tf (hunk 3 above)
Comment (bryantbiggs) [reply examples/complete-kubernetes-addons/main.tf]: Corrected in https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/551/commits/3335bd466858a2527a90dad8256a0597a0a56262

File: examples/analytics/emr-on-eks/main.tf (hunk 2 above)
Comment (kcoleman731) [reply examples/analytics/emr-on-eks/main.tf:138]: Gotcha...more specifically I'm referring to the azs and private subnets props. I.E. https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/examples/complete-kubernetes-addons/main.tf#L50

Comment (bryantbiggs) [reply examples/analytics/emr-on-eks/main.tf:138]: I will revert

**Comment by bryantbiggs** [comment]
e2e is mostly working - transient failures due to waiting for ENIs https://github.com/aws-ia/terraform-aws-eks-blueprints/runs/6617075905?check_suite_focus=true

**Comment by kbumsik** [comment]
I really like this idea and thanks for recognizing and solving the problem of `aws-resouce-tags`.

I just wanted to leave a small feedback that this indeed breaks compatibility of existing Terraform code and resources, meaning it is not a just a bugfix like v4.0.6 -> v4.0.7. So please consider bumping up at least the minor version number to follow semver.

**Comment by schwichti** [comment]
To be honest, I never fully understood the rationale behind the org-tenant-environment-zone naming scheme.

**Pull request merged**